hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java

```diff
@@ -235,6 +238,10 @@ public void verifyAndFixupContainerData(ContainerData containerData)
         return;
       }
 
+      if (!isMatchedLastLoadedECContainer(kvContainer, containerSet.getContainerMetadataStore())) {
+        return;
```

Comment:
So are we saying we would not load the container if the replica index doesn't match. This would be wrong since we need to add the container even if the replica index doesn't match. As discussed offline we should add the container to in memory container set irrespective of whether the replica index matches or not and not update the container Set store on disk. I believe it would be good if the logic can be kept inside container set.

Reply:
What I was suggesting is in the absence of a replica index matching we want to still support loading one of the containers with non matching replica index. IMO we should atleast load that container in memory by not updating replica index on disk. From what I understand we can still do that without the refactoring. I am not really sure about what refactoring you are talking about.

Reply:
update the conclusion, as this JIRA task is to avoid unknown container load, and there is no chance of having unknown index as check is already present when the container is loaded.